hca/dss/util/__init__.py

```diff
@@ -1,3 +1,39 @@
+import os
+from scandir import scandir
+
+
 def separator_to_camel_case(separated, separator):
     components = separated.split(separator)
     return "".join(x.title() for x in components)
+
+
+def directory_builder(src_dir):
+    """
+    Function that recursively locates files within folder
+    Note: scandir does not guarantee ordering
+    :param src_dir:  string for directory to be parsed through
+    :return an iterable of DirEntry objects all files within the src_dir
+    """
+    for x in scandir(os.path.join(src_dir)):
+        if x.is_dir(follow_symlinks=False):
+            for x in directory_builder(x.path):
+                yield x
+        else:
+            yield x
+
+
+def object_name_builder(file_name, src_dir):
```

Comment:
Can you explain the difference between this implementation and:
```
def object_name_builder(file_name, src_dir):
    return os.path.normpath(os.path.join(file_name)).replace(src_dir, "")
```

Reply:
i like your implementation more, ill refactor.